src/java.base/share/classes/java/net/SocksSocketImpl.java

```diff
@@ -297,17 +299,16 @@ public ProxySelector run() {
             URI uri;
             // Use getHostString() to avoid reverse lookups
             String host = epoint.getHostString();
-            // IPv6 literal?
-            if (epoint.getAddress() instanceof Inet6Address &&
-                (!host.startsWith("[")) && (host.indexOf(':') >= 0)) {
+            if (isIPv6LiteralAddress(host)) {
                 host = "[" + host + "]";
+            } else {
+                host = ParseUtil.encodePath(host);
             }
+            // IPv6 literal?
```

Comment:
Hello Daniel, this comment seems to have been misplaced. I think it should have been a few lines above just when `isIPv6LiteralAddress` is being called.

Reply:
Good catch, this comment doesn't make much sense here. `isIPv6LiteralAddress` is descriptive enough, I'll just remove the comment.